src/Templates/ServiceClientTemplate.cshtml

```diff
@@ -1,125 +1,66 @@
-﻿@using AutoRest.Go
-@using AutoRest.Core.Model
+﻿@using AutoRest.Core.Model
+@using AutoRest.Core.Utilities
+@using AutoRest.Go
 @using AutoRest.Go.Templates
 @using AutoRest.Go.Model
-@using AutoRest.Core.Utilities
 @using System.Linq
 
 @inherits AutoRest.Core.Template<AutoRest.Go.Model.CodeModelGo>
+
 @{
-    var imports = Model.ClientImports;
-    var modelName = CodeNamerGo.Instance.CamelCase(Model.Name);
-    var methods = Model.ClientMethods.OrderBy(m => m.Name.Value);
+    var ctorPrefix = "New";
+    var ctorSuffix = Model.BaseClient;
+    if (!CodeNamerGo.Instance.ExportClientTypes)
+    {
+        ctorPrefix = "new";
+        ctorSuffix = Model.BaseClient.ToPascalCase();
+    }
+}
+
+@if (CodeNamerGo.Instance.ExportClientTypes)
+{
+    WrapComment("// ", Model.GetDocumentation());
 }
-@WrapComment("// ", Model.GetDocumentation())
 package @Model.Namespace
 @EmptyLine
 @Header("// ")
 
 @EmptyLine
 import (
-@foreach (var import in imports)
-{
-    @:@(import)
-}
+    "net/url"
+    "@CodeNamerGo.Instance.PipelineImportPath"
 )
 
 @EmptyLine
 const (
-@if (!Model.IsCustomBaseUri)
-{
-<text>
-    // DefaultBaseURI is the default URI used for the service @(Model.ServiceName)
-    DefaultBaseURI = "@Model.BaseUrl"
-</text>
-}
-    @(Model.ConstGlobalDefaultParameters)
+    // ServiceVersion specifies the version of the operations used in this package.
+    ServiceVersion = "@(Model.ApiVersion)"
 )
 
 @EmptyLine
 @WrapComment("// ", Model.ClientDocumentation)
 type @(Model.BaseClient) struct {
-    autorest.Client
-@if (!Model.IsCustomBaseUri)
-{
-    @:BaseURI string
-}
-@foreach (var p in Model.Properties)
-{
-    if (!p.SerializedName.IsApiVersion())
-    {
-            @:@(string.Format((p.IsRequired || p.ModelType.CanBeEmpty()
-                                                  ? "{0} {1}" : "{0} *{1}"),
-                                                  p.Name, p.ModelType.Name))
-    }
-}
+    url url.URL
```

Comment:
>url url.UR [](start = 4, length = 10)

If this url is replacing the BaseURI, and there is no way to change it, sovereign cloud users will have a bad time

Reply:
This will be addresses when I get the v3 generator working for ARM.